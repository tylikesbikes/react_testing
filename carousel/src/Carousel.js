import { useState } from "react";
import "./Carousel.css";
import Card from "./Card";


/** Carousel: displays images and arrows to navigate through them
 * 
 * Props:
 * - photos: array of {src, caption} objects
 * - title: string describing the collection of images
 * 
 * State:
 * - currCardIdx: integer for current card index
 * 
 * App --> Carousel --> Card
 */
 function Carousel({ photos, title }) {
  const [currCardIdx, setCurrCardIdx] = useState(0);
  const [leftIsShowing, setLeftIsShowing] = useState(false);
  const [rightIsShowing, setRightIsShowing] = useState(true);

  const currCard = photos[currCardIdx];
  const total = photos.length;

  function updateArrows(idx) {
    /*set defaults = visible*/
    setLeftIsShowing(true);
    setRightIsShowing(true);

    /*hide arrows based on currCardIdx*/
    if (idx === 0) {
      setLeftIsShowing(false);
    }
    if (idx === total-1) {
      setRightIsShowing(false)
    }
  }


  function goForward() {
    let idx = 0;
    setCurrCardIdx(currCardIdx+1);
    idx = currCardIdx+1;
    
    updateArrows(idx);
  }

  
  function goBack() {
    let idx = 0;
    setCurrCardIdx(currCardIdx-1);
    idx = currCardIdx-1;
    updateArrows(idx);
  }

  return (
    <div className="Carousel">
      <h1>{title}</h1>
      <div className="Carousel-main">
        <i
          className="bi bi-arrow-left-circle"
          onClick={goBack}
          style={{visibility: leftIsShowing ? "visible":"hidden"}}
          
        />
        <Card
          caption={currCard.caption}
          src={currCard.src}
          currNum={currCardIdx + 1}
          totalNum={total}
        />
        <i
          className="bi bi-arrow-right-circle"
          onClick={goForward}
          style={{visibility: rightIsShowing ? "visible":"hidden"}}
        />
      </div>
      <p>ctr new test</p>
    </div>
  );
}

export default Carousel;
