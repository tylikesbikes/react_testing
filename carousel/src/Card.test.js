import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";


render(<Carousel   
    photos = {TEST_IMAGES}
    title="images for testing"
    />);

/* smoke test for Card */
it ('passes the basic rendering smoke test -- Card', function() {
  render(<Card 
    caption={'testCaption'}
    src={TEST_IMAGES[0].src}
    currNum={0}
    totalNum={3}
    />);
});

/*snapshop test for Card */
it('should match snapshot', () => {
    const { asFragment } = render(<Card     
    caption={'testCaption'}
    src={TEST_IMAGES[0].src}
    currNum={0}
    totalNum={3}/>);

    expect(asFragment()).toMatchSnapshot();
});