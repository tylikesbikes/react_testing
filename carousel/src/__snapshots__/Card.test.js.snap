// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testCaption
    </h4>
    <img
      alt="testCaption"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image 0 of 3.
    </small>
  </div>
</DocumentFragment>
`;
